common/beerocks/bwl/dwpal/mon_wlan_hal_dwpal.cpp

```diff
@@ -677,94 +677,92 @@ bool mon_wlan_hal_dwpal::update_vap_stats(const std::string &vap_iface_name, SVa
-         "LastDataUplinkRate=", 0},
-
-        /* Must be at the end */
-        {NULL, NULL, DWPAL_NUM_OF_PARSING_TYPES, NULL, 0}};
-
-    if (dwpal_string_to_struct_parse(reply, replyLen, fieldsToParse, sizeof(SStaStats)) ==
-        DWPAL_FAILURE) {
-        LOG(ERROR) << "DWPAL parse error ==> Abort";
+    // RSSI
+    if (!read_param_str("ShortTermRSSIAverage", reply, &tmp_str)) {
+        LOG(ERROR) << "Failed reading ShortTermRSSIAverage parameter!";
         return false;
     }
-
-    /* TEMP: Traces... */
-    // LOG(DEBUG) << "GET_STA_MEASUREMENTS reply= \n" << reply;
-    // LOG(DEBUG) << "numOfValidArgs[0]= " << numOfValidArgs[0] << " BytesSent= " << BytesSent;
-    // LOG(DEBUG) << "numOfValidArgs[1]= " << numOfValidArgs[1] << " BytesReceived= " << BytesReceived;
-    // LOG(DEBUG) << "numOfValidArgs[2]= " << numOfValidArgs[2] << " PacketsSent= " << PacketsSent;
-    // LOG(DEBUG) << "numOfValidArgs[3]= " << numOfValidArgs[3] << " PacketsReceived= " << PacketsReceived;
-    // LOG(DEBUG) << "numOfValidArgs[4]= " << numOfValidArgs[4] << " RetransCount= " << sta_stats.retrans_count;
-    // LOG(DEBUG) << "numOfValidArgs[5]= " << numOfValidArgs[5] << " ShortTermRSSIAverage= " << ShortTermRSSIAverage;
-    // LOG(DEBUG) << "numOfValidArgs[6]= " << numOfValidArgs[6] << " SNR= " << SNR;
-    // LOG(DEBUG) << "numOfValidArgs[0]= " << numOfValidArgs[7] << " Active= " << Active;
-    // LOG(DEBUG) << "numOfValidArgs[7]= " << numOfValidArgs[8] << " LastDataDownlinkRate= " << LastDataDownlinkRate;
-    // LOG(DEBUG) << "numOfValidArgs[8]= " << numOfValidArgs[9] << " LastDataUplinkRate= " << LastDataUplinkRate;
-    /* End of TEMP: Traces... */
-
-    for (uint8_t i = 0; i < (sizeof(numOfValidArgs) / sizeof(size_t)); i++) {
-        if (numOfValidArgs[i] == 0) {
-            LOG(ERROR) << "Failed reading parsed parameter " << (int)i << " ==> Abort";
-            return false;
-        }
-    }
-
-    //save average RSSI in watt
-    // ShortTermRSSIAverage values are space-separated:
-    for (const auto &rssi : beerocks::string_utils::str_split(ShortTermRSSIAverage, ' ')) {
-        float s_float = float(beerocks::string_utils::stoi(std::string(rssi)));
+    // Format "ShortTermRSSIAverage=%d %d %d %d"
+    auto samples = beerocks::string_utils::str_split(tmp_str, ' ');
+    for (const auto &s : samples) {
+        float s_float = float(beerocks::string_utils::stoi(s));
         if (s_float > beerocks::RSSI_MIN) {
             sta_stats.rx_rssi_watt += std::pow(10, s_float / float(10));
             sta_stats.rx_rssi_watt_samples_cnt++;
         }
     }
 
-    //save average SNR in watt
-    // SNR values are space-separated:
-    for (const auto &snr : beerocks::string_utils::str_split(SNR, ' ')) {
-        float s_float = float(beerocks::string_utils::stoi(std::string(snr)));
-        if (s_float > beerocks::SNR_MIN) {
+    // SNR
+    if (!read_param_str("SNR", reply, &tmp_str)) {
```

Comment:
(nitpick, two times)
- don't use tmp_str, but give it a proper name and use it only once;
- declare just before using it.
```suggestion
    std::string SNR;
    if (!read_param("SNR", reply, SNR)) {
```

Reply:
Why do you against declaring temporary helper variables? It seems redundant to me to declare a new variable to each parameter we want to read. Also, it makes the code size longer, and unnecessarily increases the stack size (lesser performance).
I don't see how it even contributes to the readability.

Reply:
It's more of a personal preference and it's not mentioned in our coding style so feel free to do however you like. My reasoning:
- (bad) it makes the vertical size of the code slightly longer;
- (good) it gives meaning to the variable - not really relevant here because the "SNR" is really close to where it is used, but there is some code below where there's more distance and you do manipulations on a tmp_something that has no clear meaning;
- (good) it makes it trivial to know which value the variable has when looking at part of the code.
- (good) it allows you to reorder the code as you like - currently, because they use the same variable name, you can't keep the calc_curr_traffic calls for PacketsSent and PacketsReceived etc. together because it's always tmp_int;
- (neutral) it makes it slightly easier for the compiler to optimize the code - not really relevant for us because gcc converts everything to full SSA, i.e. every assignment already creates a new variable;
- (good) in the light of the above: it can make debugging easier because there's a bigger chance the debug info will be able to point to that specific variable.